Dockerfile

```diff
@@ -2,27 +2,26 @@
 FROM node:11-alpine AS base
 # set working directory
 WORKDIR /app
-# copy project file
-COPY package*.json ./
-
-# ---- Dependencies ----
-FROM base AS dependencies
-# install node packages
-RUN npm set progress=false
-# install ALL node_modules, including 'devDependencies'
-RUN npm install
 
-# ---- Test ----
-## run linters, setup and tests
-FROM dependencies AS test
+# ---- Development ----
+FROM base as development
+COPY package*.json ./
+RUN npm ci --production
```

Comment:
What is your reasoning behind doing the prod `npm ci` here?

I can see that these get shelved and then used in the `release` build, but would it not be more straight-forward to have this get run in that build instead?

Reply:
I believe the logic here is to have the absolute bare minimum in the release image as well as making the build as quick as possible. Doing it this way I guess means the installation is quicker because you have the lockfile available without needing to copy the lockfile to the release image. I suspect it really makes no odds to us but this seems to be common practise.